app/validators/UpdateProfileValidator.ts

```diff
@@ -0,0 +1,21 @@
+import { schema, CustomMessages, rules } from '@ioc:Adonis/Core/Validator'
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+export default class UpdateProfileValidator {
+  constructor(protected ctx: HttpContextContract) {}
+  public schema = schema.create({
+    name: schema.string.optional({}, [rules.minLength(3), rules.maxLength(30)]),
+    gender: schema.enum.optional(['MALE', 'FEMALE']),
+    dob: schema.date.optional(),
+    mobile: schema.string.optional({}, [
+      rules.unique({ table: 'profile', column: 'mobile' }),
+      rules.mobile(),
+      rules.maxLength(10),
+      rules.minLength(10),
```

Comment:
not require

Reply:
fixed

Reply:
@sombirCredilio  no need to add length validation explicitly. Length check is being performed inside rule.mobile()